torch/_functorch/aot_autograd.py

```diff
@@ -1577,9 +1605,9 @@ class AOTConfig:
 def maybe_to_fresh_input(idx, t, meta):
     if not isinstance(t, Tensor):
         return t
-    if idx in meta.mutated_inp_indices:
+    if idx in meta.mutated_inp_runtime_indices:
         # We only need to bother cloning mutated inputs that participate in autograd.
-        mutated_inp_idx = meta.mutated_inp_indices.index(idx)
+        mutated_inp_idx = meta.mutated_inp_runtime_indices.index(idx)
         if meta.input_info[idx].requires_grad and meta.input_info[idx].mutates_data:
```

Comment:
Is this requires_grad check necessary anymore? I think this codepath can now be:
```
if meta.input_info[idx].mutates_data:
    return t.clone()
elf meta.input_info[idx].mutates_metadata:
    return t.view(t.shape)
```

Reply:
Nvm I think it still is: because we're saying that if we have subclass input that does not require grad but gets mutated, we will still not keep it in the graph (but we don't want to clone() it here)